plugin/src/automation/search/equivalence.mli

```diff
@@ -9,10 +9,10 @@ open Stateutils
 
 (*
  * Prove section and retraction
- * Return the section term and the retraction term
- * (Don't return the types, since Coq can infer them without issue)
+ * Return the section term and the retraction term, as well as their types
  *)
-val prove_equivalence : env -> evar_map -> lifting -> (types * types)
+val prove_equivalence :
+  env -> evar_map -> lifting -> ((constr * types) * (constr * types))
```

Comment:
Kind of inconsistent to do this for equivalence but not for adjoint_retraction or adjunction. So maybe just include the types for those too